### DISCUSSION TITLE
Underlining in syntax higlighting of code-blocks in dark mode

### DISCUSSION THREAD

**Discussion by MiroPalmu** [pradyunsg/furo#736] [Potential Issue] [answered]
Hi

Is it intended that with default options syntax highlighting for class names contains underlining in dark mode? I'm asking because underscores will overlap with this underlining. Example:

```
.. code-block:: cpp

   class some_class_name_with_underscores{};
```

![image](https://github.com/pradyunsg/furo/assets/104301888/4975a27e-124c-4ff7-906b-10be34295722)

**Comment by dgw** [comment]
~~I believe that just means the default Pygments theme(s) contains the underline, and Furo doesn't override the Pygments default.~~ You believe _wrong_, past-me! Furo sets the default `pygments_dark_style`, because that's a setting that Furo invented.

The underlining in odd places might be one reason we changed our Sphinx config for _[a project I won't self-promote]_ to explicitly use the "friendly" theme for light mode and "monokai" for dark, besides liking the colors better.

<img width="400" alt="image" src="https://github.com/pradyunsg/furo/assets/164140/ca0f8a93-fe83-433e-ada6-668330781e3e">

~~Furo changing this would kind of surprise me, since it's not actually a Furo default but a Pygments default causing it.~~

**Reply by MiroPalmu** [reply]
Ok. Thanks for the tip. For reference [here](https://pradyunsg.me/furo/customisation/colors/#code-block-styling) is documentation for how to change it, and [here](https://pygments.org/styles/) is list of builtin styles in Pygments.

> Furo doesn't override the Pygments default.

I'm new to using Furo so this might be wrong, but it seems that it's at least doing something since:

![image](https://github.com/pradyunsg/furo/assets/104301888/e09b68e7-760a-4800-bc81-42101377921a)

**Reply by dgw** [reply, marked as answer]
Huh, I stand corrected. Furo sets "native" as the default dark-mode code theme, and that's where the underline comes from.

https://github.com/pradyunsg/furo/blob/d14286c8345cc5509bc8363d15c77a4607529f74/src/furo/__init__.py#L376-L378

And `pygments_dark_style` isn't a default option in Sphinx, either:

https://github.com/pradyunsg/furo/blob/d14286c8345cc5509bc8363d15c77a4607529f74/docs/changelog.md?plain=1#L368

So I must apologize. This particular style (the dark mode highlighting) _is_ a Furo default that could be changed!
